apps/condo/domains/property/components/BasePropertyForm/index.tsx

```diff
@@ -20,20 +22,59 @@ interface IPropertyFormProps {
     type: string
 }
 
-export const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
+// Todo(zuch): wait for next.js implement router abort method and remove custom error
+// https://github.com/vercel/next.js/issues/2476
+
+const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
     const intl = useIntl()
     const AddressLabel = intl.formatMessage({ id: 'pages.condo.property.field.Address' })
     const FieldIsRequiredMsg = intl.formatMessage({ id: 'FieldIsRequired' })
     const NameMsg = intl.formatMessage({ id: 'pages.condo.property.form.field.Name' })
+
+    const ModalTitle = intl.formatMessage({ id: 'pages.condo.warning.modal.Title' })
+    const SaveLabel = intl.formatMessage({ id: 'pages.condo.warning.modal.SaveLabel' })
+    const LeaveLabel = intl.formatMessage({ id: 'pages.condo.warning.modal.LeaveLabel' })
+    const HelpMessage = intl.formatMessage({ id: 'pages.condo.warning.modal.HelpMessage' })
+
     const { action, initialValues } = props
+    const router = useRouter()
+
+    const [isMapChanged, setIsMapChanged] = useState(false)
+    const [next, setNext] = useState(null)
+    const [isModalVisible, setIsModalVisible] = useState(false)
+    const showModal = () => setIsModalVisible(true)
+    const closeModal = () => {
+        setIsMapChanged(false)
+        setIsModalVisible(false)
+    }
+    const handleCancel = () => {
+        closeModal()
+        router.push(next)
+    }
+
+    useEffect(() => {
```

Comment:
Please, create hook/component for common usage with all forms.

Reply:
Done